tests/by-util/test_cksum.rs

```diff
@@ -663,3 +663,146 @@ fn test_conflicting_options() {
         )
         .code_is(1);
 }
+
+#[test]
+fn test_check_algo_err() {
+    let scene = TestScenario::new(util_name!());
+
+    let at = &scene.fixtures;
+
+    at.touch("f");
+
+    scene
+        .ucmd()
+        .arg("--a")
+        .arg("sm3")
+        .arg("--check")
+        .arg("f")
+        .fails()
+        .no_stdout()
+        .stderr_contains("cksum: no properly formatted checksum lines found")
+        .code_is(1);
+}
+
+#[test]
+fn test_cksum_check() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    let commands = [
+        vec!["-a", "sha384"],
+        vec!["-a", "blake2b"],
+        vec!["-a", "blake2b", "-l", "384"],
+        vec!["-a", "sm3"],
+    ];
+    at.touch("f");
+    at.touch("CHECKSUM");
+    for command in &commands {
+        let result = scene.ucmd().args(command).arg("f").succeeds();
+        at.append("CHECKSUM", result.stdout_str());
```

Comment:
If the checksum computation ever breaks, it likely affects both checksum computation and verification in the same way, and therefore would not cause this test to fail.

I suggest just precomputing the file `CHECKSUM` and adding it to the fixture directory in the git repo (perhaps under a different name). This way, we will definitely notice if anything ever goes wrong.

Reply:
Thank you!